staging/src/k8s.io/apimachinery/pkg/runtime/serializer/json/json.go

```diff
@@ -242,10 +254,70 @@ func (s *Serializer) doEncode(obj runtime.Object, w io.Writer) error {
 		_, err = w.Write(data)
 		return err
 	}
+
+	if s.streamingEncoder {
+		typeMeta, listMeta, items, err := meta.GetListMeta(obj)
+		if err == nil {
+			return streamingEncodeList(w, typeMeta, listMeta, items)
+		}
+	}
 	encoder := json.NewEncoder(w)
 	return encoder.Encode(obj)
 }
 
+func streamingEncodeList(w io.Writer, typeMeta v1.TypeMeta, listMeta v1.ListMeta, items []runtime.Object) error {
+	_, err := w.Write([]byte(`{`))
+	if err != nil {
+		return err
+	}
+	if typeMeta.Kind != "" {
+		_, err = w.Write([]byte(fmt.Sprintf(`"kind":%q,`, typeMeta.Kind)))
```

Comment:
I think the compiler is free to generate, eg <tt>"kind":\`ConfigMap\`</tt> here.

We've got unit tests, but I wonder how we can be really sure that the difference between Golang string literals and JSON string literals will never cause problems.

Reply:
Right, fmt was left from PoC. Fixed it to use json marshaller.